vmm/src/lib.rs

```diff
@@ -1081,6 +1173,149 @@ impl Vmm {
+                    } else {
+                        warn!("Configuration not sent yet");
+                        Response::error().write_to(&mut socket)?;
+                    }
+                }
+                Command::MemoryFd => {
+                    info!("MemoryFd Command Received");
+
+                    if !started {
+                        warn!("Migration not started yet");
+                        Response::error().write_to(&mut socket).map_err(|e| {
+                            MigratableError::MigrateReceive(anyhow!(
+                                "Error sending response: {}",
+                                e
+                            ))
+                        })?;
+                    }
+
+                    match &mut socket {
+                        SocketStream::Unix(unix_socket) => {
+                            // For Unix sockets, we can receive file descriptors
+                            let mut buf = [0u8; 4];
+                            let (_, file) = unix_socket.recv_with_fd(&mut buf).map_err(|e| {
+                                MigratableError::MigrateReceive(anyhow!(
+                                    "Error receiving slot from socket: {}",
+                                    e
+                                ))
+                            })?;
+
+                            if existing_memory_files.is_none() {
+                                existing_memory_files = Some(HashMap::new())
+                            }
+
+                            if let Some(ref mut existing_memory_files) = existing_memory_files {
+                                let slot = u32::from_le_bytes(buf);
+                                existing_memory_files.insert(slot, file.unwrap());
+                            }
+
+                            Response::ok().write_to(&mut socket).map_err(|e| {
+                                MigratableError::MigrateReceive(anyhow!(
+                                    "Error sending response: {}",
+                                    e
+                                ))
+                            })?;
+                        }
+                        SocketStream::Tcp(_tcp_socket) => {
+                            // For TCP sockets, we cannot transfer file descriptors
+                            warn!(
+                                "MemoryFd command received over TCP socket, which is not supported"
+                            );
+                            Response::error().write_to(&mut socket).map_err(|e| {
+                                MigratableError::MigrateReceive(anyhow!(
+                                    "Error sending response: {}",
+                                    e
+                                ))
+                            })?;
+                            return Err(MigratableError::MigrateReceive(anyhow!(
+                                "MemoryFd command not supported over TCP sockets"
+                            )));
```

Comment:
Since `send_memory_fd` would (and should) never be invoked on `TcpSocket`, should we change the whole section to `unreachable!("MemoryFd command not supported over TCP sockets")`

Reply:
> Since `send_memory_fd` would (and should) never be invoked on `TcpSocket`, should we change the whole section to `unreachable!("MemoryFd command not supported over TCP sockets")`

You are right. TCP does not use MemoryFd, so I will remove the TCP related code here.